### DISCUSSION TITLE
dictionary_word_is_known functionality

### DISCUSSION THREAD

**Discussion by as-cal8** [opencog/link-grammar#1474] [Q&A] [answered]
Hello,
I just started using LGP and I wanted to check before parsing anything if some special words are not known by the dictionary. Therefore I used the function "`dictionary_word_is_known`" but it returns `true` for some random words.

Setup from "hello world example" using 5.12.3 release:

```
   Log            log;
   Dictionary    dict;
   Parse_Options opts;
   Sentence      sent;
   Linkage       linkage;
   char *        diagram;
   int           i, num_linkages, status;

   setlocale(LC_ALL, "");
   opts = parse_options_create(); // parse options, can be changed with routines like parse_options_set_min_null_count, parse_options_set_max_null_count
   dict = dictionary_create_lang("en");
   if (!dict) {
      printf ("Fatal error: Unable to open the dictionary\n");
      return 1;
   }

   char* rnd = {"SHAldkfjgjh\0"};
   bool is_known = dictionary_word_is_known(dict, rnd);
   assert(is_known);
   rnd = {"SHALALA\0"};
   is_known = dictionary_word_is_known(dict, rnd);
   assert(is_known);
   rnd = {"SHALL\0"};
   is_known = dictionary_word_is_known(dict, rnd);
   assert(is_known);
   rnd = {"shalwerkjALALA\0"};
   is_known = dictionary_word_is_known(dict, rnd);
   assert(is_known);
```

Executing the above causes it to fail only at the last assert statement.

Is something wrong with setting up the dict?
Also when parsing a simple sentence containing "SHALL" it is recognized as a noun even though there is not a noun like that to my knowledge and its not contained in the dictionary in upp case format. 
I expected it to be case sensitive and causing an error when parsing smthing like "This guy SHALL provide the next sentence."
Instead thats the result (using default parse options same as in hello world example):

![image](https://github.com/opencog/link-grammar/assets/78380441/ca43b4d5-421f-45e0-a60e-c92c8d153460)

Where the links connecting SHALL dont make sense to me.

Would appreciate any help here.
Best regards, Alex

**Comment by linas** [comment, marked as answer]
You may have "unknown word" defined. This enables rules that attempt to guess unknown words, based on the form. For example, there are recognizers for various Latinate ends, common in science lit. There are also regexes for roman numerals, and some other specialty forms. The above example used the `<ALL-UPPER>` regex. 

Surprisingly, there is no C API to turn it on and off (perhaps it should be added?)  It can be turned off by commenting out `UNKNOWN-WORD` in the dictionary.  Might also need to disable the `<ALL-UPPER>` regex as well? Not sure. 

Even if  `UNKNOWN-WORD`  is disabled, the spell-guesser might run. That can be disabled with the `parse_options_set_spell_guess()` API call.
